web/pgadmin/static/js/SchemaView/index.jsx

```diff
@@ -773,7 +773,7 @@ function SchemaDialogView({
               <DefaultButton data-test="Reset" onClick={onResetClick} startIcon={<SettingsBackupRestoreIcon />} disabled={!dirty || saving} className={classes.buttonMargin}>
                 {gettext('Reset')}
               </DefaultButton>
-              <PrimaryButton data-test="Save" onClick={onSaveClick} startIcon={ButtonIcon} disabled={ !(viewHelperProps.mode === 'edit' ? dirty : true) || saving || Boolean(formErr.name) || !formReady}>
+              <PrimaryButton data-test="Save" onClick={onSaveClick} startIcon={ButtonIcon} disabled={ !(viewHelperProps.mode === 'edit' || viewHelperProps.mode == 'create' ? dirty : true) || saving || Boolean(formErr.name) || !formReady}>
```

Comment:
No this is not the correct fix. If a data is valid then `Save` button should be enabled irrespective of dirty or not.

Reply:
The correct fix for this would be to use `edit` mode in filter dialog. And handle things in backend accordingly.